Remove support for Python 3.7 in the Github Actions test matrix

Required Python packages no longer install on Python 3.7.

```
Run python -m pip install --upgrade pip
  python -m pip install --upgrade pip
  python -m pip install -r .ci/dev-requirements.txt --cache-dir pipcache
  python -m ipykernel install --user --name openvino_env
  shell: /usr/bin/bash -e {0}
  env:
    PIP_CACHE_KEY: cache-pip-[2](https://github.com/openvinotoolkit/openvino_notebooks/actions/runs/7366633282/job/20049085792#step:7:2)02[3](https://github.com/openvinotoolkit/openvino_notebooks/actions/runs/7366633282/job/20049085792#step:7:3).0.0
    FILES_CACHE_KEY: files-cache-2022-03-23
    USER_CACHE_KEY: user-cache-2022-03-23
    HUB_HOME: /tmp/paddlehub
    pythonLocation: /opt/hostedtoolcache/Python/3.7.17/x6[4](https://github.com/openvinotoolkit/openvino_notebooks/actions/runs/7366633282/job/20049085792#step:7:4)
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.7.17/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.7.17/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.7.17/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.7.17/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.7.17/x64/lib
Requirement already satisfied: pip in /opt/hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages (23.3.2)
Looking in indexes: https://pypi.org/simple, https://download.pytorch.org/whl/cpu
Ignoring tensorflow-macos: markers 'sys_platform == "darwin" and platform_machine == "arm64" and python_version > "3.8"' don't match your environment
Ignoring tensorflow-macos: markers 'sys_platform == "darwin" and platform_machine == "arm64" and python_version <= "3.8"' don't match your environment
Ignoring tensorflow: markers 'sys_platform == "darwin" and platform_machine != "arm64" and python_version > "3.8"' don't match your environment
Ignoring tensorflow: markers 'sys_platform == "darwin" and platform_machine != "arm64" and python_version <= "3.8"' don't match your environment
Ignoring tensorflow: markers 'sys_platform != "darwin" and python_version > "3.8"' don't match your environment
Collecting openvino-dev>=2023.2.0 (from -r .ci/../requirements.txt (line 2))
  Using cached openvino_dev-2023.2.0-13089-py3-none-any.whl.metadata (17 kB)
Collecting openvino-telemetry>=2023.2.1 (from -r .ci/../requirements.txt (line 3))
  Downloading openvino_telemetry-2023.2.1-py3-none-any.whl.metadata (2.3 kB)
Collecting nncf==2.7.0 (from -r .ci/../requirements.txt (line 4))
  Using cached nncf-2.7.0-py3-none-any.whl.metadata (24 kB)
Collecting tensorflow<=2.12.0,>=2.[5](https://github.com/openvinotoolkit/openvino_notebooks/actions/runs/7366633282/job/20049085792#step:7:5) (from -r .ci/../requirements.txt (line 13))
  Downloading tensorflow-2.11.0-cp37-cp37m-manylinux_2_17_x8[6](https://github.com/openvinotoolkit/openvino_notebooks/actions/runs/7366633282/job/20049085792#step:7:6)_64.manylinux2014_x86_64.whl (588.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 588.3/588.3 MB 4.9 MB/s eta 0:00:00
Collecting onnx>=1.11.0 (from -r .ci/../requirements.txt (line 15))
  Downloading onnx-1.14.1-cp3[7](https://github.com/openvinotoolkit/openvino_notebooks/actions/runs/7366633282/job/20049085792#step:7:7)-cp37m-manylinux_2_17_x[8](https://github.com/openvinotoolkit/openvino_notebooks/actions/runs/7366633282/job/20049085792#step:7:8)6_64.manylinux2014_x86_64.whl.metadata (15 kB)
ERROR: Ignored the following versions that require a different python version: 1.15.0 Requires-Python >=3.8
ERROR: Could not find a version that satisfies the requirement torch==2.1.0 (from versions: 0.4.1, 0.4.1.post2, 1.0.0, 1.0.1, 1.0.1.post2, 1.1.0, 1.2.0, 1.2.0+cpu, 1.3.0, 1.3.0+cpu, 1.3.1, 1.3.1+cpu, 1.4.0, 1.4.0+cpu, 1.5.0, 1.5.0+cpu, 1.5.1, 1.5.1+cpu, 1.6.0, 1.6.0+cpu, 1.7.0, 1.7.0+cpu, 1.7.1, 1.7.1+cpu, 1.8.0, 1.8.0+cpu, 1.8.1, 1.8.1+cpu, 1.[9](https://github.com/openvinotoolkit/openvino_notebooks/actions/runs/7366633282/job/20049085792#step:7:9).0, 1.9.0+cpu, 1.9.1, 1.9.1+cpu, 1.[10](https://github.com/openvinotoolkit/openvino_notebooks/actions/runs/7366633282/job/20049085792#step:7:10).0, 1.10.0+cpu, 1.10.1, 1.10.1+cpu, 1.10.2, 1.10.2+cpu, 1.[11](https://github.com/openvinotoolkit/openvino_notebooks/actions/runs/7366633282/job/20049085792#step:7:11).0, 1.11.0+cpu, 1.[12](https://github.com/openvinotoolkit/openvino_notebooks/actions/runs/7366633282/job/20049085792#step:7:12).0, 1.12.0+cpu, 1.12.1, 1.12.1+cpu, 1.[13](https://github.com/openvinotoolkit/openvino_notebooks/actions/runs/7366633282/job/20049085792#step:7:13).0, 1.13.0+cpu, 1.13.1, 1.13.1+cpu)
ERROR: No matching distribution found for torch==2.1.0
Error: Process completed with exit code 1.
```

https://github.com/openvinotoolkit/openvino_notebooks/actions/runs/7366673425/job/20049150643?pr=1586#step:7:36

--- a/.github/workflows/codecheck.yml
+++ b/.github/workflows/codecheck.yml
@@ -32,19 +32,19 @@ jobs:
     runs-on: ubuntu-20.04  # change cachepip step when changing this
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v2
+      uses: actions/checkout@v4
     - name: Dotenv Action
       id: dotenv
       uses: xom9ikk/dotenv@v1.0.2
       with:
         path: ./.github/workflows
     - name: Set up Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v5
       with:
         python-version: 3.8  # change cachepip step when changing this
     - name: Cache OpenVINO Pip Packages
       id: cachepip
-      uses: actions/cache@v2
+      uses: actions/cache@v3
       with:
         path: |
           pipcache

--- a/.github/workflows/codeql.yml
+++ b/.github/workflows/codeql.yml
@@ -22,9 +22,9 @@ jobs:
 
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
       - name: Set up Python
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v5
         with:
           python-version: '3.7'
       - name: Install dependencies
@@ -37,7 +37,7 @@ jobs:
           # that includes the dependencies
           echo "CODEQL_PYTHON=$(which python)" >> $GITHUB_ENV
       - name: Initialize CodeQL
-        uses: github/codeql-action/init@v1
+        uses: github/codeql-action/init@v3
         with:
           languages: python
           # Override the default behavior so that the action doesn't attempt
@@ -45,5 +45,5 @@ jobs:
           setup-python-dependencies: false
           
       - name: Perform CodeQL Analysis
-        uses: github/codeql-action/analyze@v1
+        uses: github/codeql-action/analyze@v3
 

--- a/.github/workflows/convert_notebooks.yml
+++ b/.github/workflows/convert_notebooks.yml
@@ -59,14 +59,14 @@ jobs:
     # This should ideally be a reusable workflow
 
     - name: Checkout repository
-      uses: actions/checkout@v2
+      uses: actions/checkout@v4
     - name: Dotenv Action
       id: dotenv
       uses: xom9ikk/dotenv@v1.0.2
       with:
         path: ./.github/workflows
     - name: Set up Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v5
       with:
         python-version: ${{ matrix.python }}
     - name: Install required packages
@@ -78,7 +78,7 @@ jobs:
 
     - name: Cache OpenVINO Pip Packages
       id: cachepip
-      uses: actions/cache@v2
+      uses: actions/cache@v3
       with:
         path: |
           pipcache
@@ -87,7 +87,7 @@ jobs:
     # Cache specific files to reduce downloads or prevent network issues
     - name: Cache Files
       id: cachefiles
-      uses: actions/cache@v2
+      uses: actions/cache@v3
       with:
         path: |
           # NOTE: when modifying cache paths, update FILES_CACHE_KEY in .env
@@ -118,7 +118,7 @@ jobs:
     # Cache PaddlePaddle cache directories to prevent CI failing due to network/download issues
     - name: Cache PaddlePaddle cache directories (per OS)
       id: cacheusercache
-      uses: actions/cache@v2
+      uses: actions/cache@v3
       with:
         path: |
           ${{ env.HUB_HOME }}
@@ -174,9 +174,9 @@ jobs:
         python -m pip freeze
         python -m pip freeze > pip-freeze-${{ github.sha }}-${{matrix.os}}-${{ matrix.python }}.txt
     - name: Archive pip freeze
-      uses: actions/upload-artifact@v2
+      uses: actions/upload-artifact@v4
       with:
-        name: pip-freeze
+        name: pip-freeze-${{matrix.os}}-${{ matrix.python }}
         path: pip-freeze-${{ github.sha }}-${{matrix.os}}-${{ matrix.python }}.txt
 
     #### End installation/preparation
@@ -185,7 +185,7 @@ jobs:
       shell: bash
       run: .ci/convert_notebooks.sh
     - name: Save reStructuredText files
-      uses: actions/upload-artifact@v2
+      uses: actions/upload-artifact@v4
       with:
         name: rst_files
         path: rst_files

--- a/.github/workflows/docker.yml
+++ b/.github/workflows/docker.yml
@@ -50,7 +50,7 @@ jobs:
             sudo rm -rf /opt/ghc
             echo "Available storage:"
             df -h
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v4
     - name: Build Docker image
       run: |
         docker build . -t openvino_notebooks

--- a/.github/workflows/generate_tags.yml
+++ b/.github/workflows/generate_tags.yml
@@ -18,21 +18,21 @@ jobs:
     runs-on: ubuntu-20.04  # change cachepip step when changing this
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v2
+      uses: actions/checkout@v4
     - name: Dotenv Action
       id: dotenv
       uses: xom9ikk/dotenv@v1.0.2
       with:
         path: ./.github/workflows
     - name: Set up Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v5
       with:
         python-version: 3.8
     - name: Run tagger and store results in file
       run: |
         python .ci/tagger.py > notebook-tags-${{ github.sha }}.json
     - name: Archive notebook tags
-      uses: actions/upload-artifact@v2
+      uses: actions/upload-artifact@v4
       with:
         name: notebook-tags
         path: notebook-tags-${{ github.sha }}.json

--- a/.github/workflows/install_requirements.yml
+++ b/.github/workflows/install_requirements.yml
@@ -16,22 +16,22 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-20.04, ubuntu-22.04, windows-2019, windows-2022, macos-11, macos-12]
-        python: ["3.7", "3.8", "3.9", "3.10"]
+        python: ["3.8", "3.9", "3.10", "3.11"]
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v2
+      uses: actions/checkout@v4
     - name: Dotenv Action
       id: dotenv
       uses: xom9ikk/dotenv@v1.0.2
       with:
         path: ./.github/workflows
     - name: Set up Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v5
       with:
         python-version: ${{ matrix.python }}
     - name: Cache OpenVINO Pip Packages
       id: cachepip
-      uses: actions/cache@v2
+      uses: actions/cache@v3
       with:
         path: |
           pipcache
@@ -60,9 +60,9 @@ jobs:
         python -m pip freeze
         python -m pip freeze > pip-freeze-${{ github.sha }}-${{matrix.os}}-${{ matrix.python }}.txt
     - name: Archive pip freeze
-      uses: actions/upload-artifact@v2
+      uses: actions/upload-artifact@v4
       with:
-        name: pip-freeze
+        name: pip-freeze-${{matrix.os}}-${{ matrix.python }}
         path: pip-freeze-${{ github.sha }}-${{matrix.os}}-${{ matrix.python }}.txt
     - name: Check install
       run: |

--- a/.github/workflows/install_requirements_china.yml
+++ b/.github/workflows/install_requirements_china.yml
@@ -16,9 +16,9 @@ jobs:
     runs-on: ${{ matrix.os }}
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v4
     - name: Set up Python 
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v5
       with:
         python-version: ${{ matrix.python }}
     - name: Install dependencies
@@ -29,9 +29,9 @@ jobs:
         python -m ipykernel install --user --name openvino_env
         python -m pip freeze > pip-freeze-${{ github.sha }}-${{matrix.os}}-${{ matrix.python }}.txt
     - name: Archive pip freeze
-      uses: actions/upload-artifact@v2
+      uses: actions/upload-artifact@v4
       with:
-        name: pip-freeze
+        name: pip-freeze-${{matrix.os}}-${{ matrix.python }}
         path: pip-freeze-${{ github.sha }}-${{matrix.os}}-${{ matrix.python }}.txt
     - name: Test that `jupyter lab` works
       run: |

--- a/.github/workflows/labeler.yml
+++ b/.github/workflows/labeler.yml
@@ -13,7 +13,7 @@ jobs:
   labelling:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v4
     - name: Add labels (Title/Description)
       uses: github/issue-labeler@v3.1
       with:

--- a/.github/workflows/pip_conflicts_check.yml
+++ b/.github/workflows/pip_conflicts_check.yml
@@ -19,7 +19,7 @@ jobs:
     runs-on: ubuntu-20.04  # change cachepip step when changing this
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v2
+      uses: actions/checkout@v4
 
     - name: Dotenv Action
       id: dotenv
@@ -28,7 +28,7 @@ jobs:
         path: ./.github/workflows
 
     - name: Set up Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v5
       with:
         python-version: 3.8  # change cachepip step when changing this
 

--- a/.github/workflows/spellcheck.yml
+++ b/.github/workflows/spellcheck.yml
@@ -29,7 +29,7 @@ jobs:
     runs-on: ubuntu-20.04  # change cachepip step when changing this
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v2
+      uses: actions/checkout@v4
 
     - name: Dotenv Action
       id: dotenv
@@ -38,13 +38,13 @@ jobs:
         path: ./.github/workflows
 
     - name: Set up Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v5
       with:
         python-version: 3.8  # change cachepip step when changing this
 
     - name: Cache OpenVINO Pip Packages
       id: cachepip
-      uses: actions/cache@v2
+      uses: actions/cache@v3
       with:
         path: |
           pipcache

--- a/.github/workflows/treon.yml
+++ b/.github/workflows/treon.yml
@@ -31,7 +31,7 @@ jobs:
     # These steps are also copied to convert_notebooks.yml
 
     - name: Checkout repository
-      uses: actions/checkout@v2
+      uses: actions/checkout@v4
     - name: Dotenv Action
       id: dotenv
       uses: xom9ikk/dotenv@v1.0.2
@@ -44,13 +44,13 @@ jobs:
             fi
       shell: bash
     - name: Set up Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v5
       with:
         python-version: ${{ matrix.python }}
 
     - name: Cache OpenVINO Pip Packages
       id: cachepip
-      uses: actions/cache@v2
+      uses: actions/cache@v3
       with:
         path: |
           pipcache
@@ -59,7 +59,7 @@ jobs:
     # Cache specific files to reduce downloads or prevent network issues
     - name: Cache Files
       id: cachefiles
-      uses: actions/cache@v2
+      uses: actions/cache@v3
       with:
         path: |
           # NOTE: when modifying cache paths, update FILES_CACHE_KEY in .env
@@ -89,7 +89,7 @@ jobs:
     # Cache PaddlePaddle cache directories to prevent CI failing due to network/download issues
     - name: Cache PaddlePaddle cache directories
       id: cacheusercache
-      uses: actions/cache@v2
+      uses: actions/cache@v3
       with:
         path: |
           ${{ env.HUB_HOME }}
@@ -147,9 +147,9 @@ jobs:
         python -m pip freeze
         python -m pip freeze > pip-freeze-${{ github.sha }}-${{matrix.os}}-${{ matrix.python }}.txt
     - name: Archive pip freeze
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@v4
       with:
-        name: pip-freeze
+        name: pip-freeze-${{matrix.os}}-${{ matrix.python }}
         path: pip-freeze-${{ github.sha }}-${{matrix.os}}-${{ matrix.python }}.txt
 
     #### End installation/preparation
@@ -185,14 +185,14 @@ jobs:
             python .ci/validate_notebooks.py --ignore_list .ci/ignore_treon_mac.txt --report_dir test_report/${{matrix.os}}-${{ matrix.python }} --timeout 1200
       shell: bash
     - name: Archive test report
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@v4
       with:
-        name: test_report.csv
+        name: test_report.csv-${{matrix.os}}-${{ matrix.python }}
         path: test_report.csv
     - name: Archive notebook test report
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@v4
       with:
-        name: test_report
+        name: test_report-${{matrix.os}}-${{ matrix.python }}
         path: test_report/
 
     # Show the cache after running the notebooks

--- a/.github/workflows/treon_precommit.yml
+++ b/.github/workflows/treon_precommit.yml
@@ -46,7 +46,7 @@ jobs:
     # These steps are also copied to convert_notebooks.yml
 
     - name: Checkout repository
-      uses: actions/checkout@v3
+      uses: actions/checkout@v4
       with:
           fetch-depth: 0
     
@@ -78,13 +78,13 @@ jobs:
             fi
       shell: bash
     - name: Set up Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v5
       with:
         python-version: ${{ matrix.python }}
 
     - name: Cache OpenVINO Pip Packages
       id: cachepip
-      uses: actions/cache@v2
+      uses: actions/cache@v3
       with:
         path: |
           pipcache
@@ -93,7 +93,7 @@ jobs:
     # Cache specific files to reduce downloads or prevent network issues
     - name: Cache Files
       id: cachefiles
-      uses: actions/cache@v2
+      uses: actions/cache@v3
       with:
         path: |
           # NOTE: when modifying cache paths, update FILES_CACHE_KEY in .env
@@ -124,7 +124,7 @@ jobs:
     # Cache PaddlePaddle cache directories to prevent CI failing due to network/download issues
     - name: Cache PaddlePaddle cache directories
       id: cacheusercache
-      uses: actions/cache@v2
+      uses: actions/cache@v3
       with:
         path: |
           ${{ env.HUB_HOME }}
@@ -182,9 +182,9 @@ jobs:
         python -m pip freeze
         python -m pip freeze > pip-freeze-${{ github.sha }}-${{matrix.os}}-${{ matrix.python }}.txt
     - name: Archive pip freeze
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@v4
       with:
-        name: pip-freeze
+        name: pip-freeze-${{matrix.os}}-${{ matrix.python }}
         path: pip-freeze-${{ github.sha }}-${{matrix.os}}-${{ matrix.python }}.txt
 
     #### End installation/preparation
@@ -220,14 +220,14 @@ jobs:
            python .ci/validate_notebooks.py --test_list test_notebooks.txt --ignore_list .ci/ignore_treon_mac.txt --report_dir test_report/${{matrix.os}}-${{ matrix.python }}
       shell: bash
     - name: Archive test report
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@v4
       with:
-        name: test_report.csv
+        name: test_report.csv-${{matrix.os}}-${{ matrix.python }}
         path: test_report.csv
     - name: Archive notebook test report
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@v4
       with:
-        name: test_report
+        name: test_report-${{matrix.os}}-${{ matrix.python }}
         path: test_report/
 
     # Show the cache after running the notebooks

